src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2661,9 +2662,9 @@ private void PushOne()
                 
                 while (true)
                 {
-                    if (_buffer.IsEmpty)
+                    if (_buffer!.IsEmpty)
                         PullIfNeeded();
-                    else if (_buffer.Peek()!.Element == NotYetThere) // Shebang is fine, we checked that the buffer is not empty
+                    else if (_buffer.Peek()!.Element.Exception == NotYetThereException) // Shebang is fine, we checked that the buffer is not empty
```

Comment:
The fix - check the value of the `Exception`s by reference. Can't get a false positive that way.